### PR TITLE
🚀 Day04: added-join-schemas-task

### PR DESCRIPTION
**Pull request by mahabubx7**
### 🚀 Day04: added-join-schemas-task
 - [x] Added JOIN schemas for tables.
 - [x] Added JOIN table's queries.
 - [x] Added Screenshots for all query's results. 
 
_ScreenShots_ (all new)

Screenshot 2023-06-09 144930
Screenshot 2023-06-09 141545
Screenshot 2023-06-09 144601
Screenshot 2023-06-09 144633
Screenshot 2023-06-09 144732
Screenshot 2023-06-09 144757
Screenshot 2023-06-09 144822
Screenshot 2023-06-09 144850
Screenshot 2023-06-09 144914

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -105,3 +105,75 @@ WHERE o.full_name = 'Dean Winchester' AND (a.escape_attempts IS NULL OR a.escape
...
+SELECT s.name AS specialty FROM visits v
+INNER JOIN animals a ON a.id = v.animal_id
+INNER JOIN vets vt ON vt.id = v.vet_id
+INNER JOIN specializations sp ON sp.vet_id = vt.id
+INNER JOIN species s ON s.id = sp.species_id
+WHERE vt.name = 'Maisy Smith' GROUP BY s.name
+ORDER BY COUNT(*) DESC LIMIT 1;
+
+
+
```
Comment (alexander16108) [inline queries.sql]: hello there 👋🏾 the result of the following queries are incorrect please try again to return the correct result and add the screenshots to your PR description 👍🏾 

- What animal has the most visits to vets?
- How many times visits where with a vet that did not specialize in that animal's species?
- What specialty should Maisy Smith consider getting? Look for the species she gets the most.

**Review by alexander16108 — CHANGES_REQUESTED** [review]
Hi @mahabubx7 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!




## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql (hunk 1 above)
Comment (mahabubx7) [reply queries.sql]: @alexander16108 
can you help me with the correct results. 
Like I'm getting visit_count = 2 for `What animal has the most visits to vets?`.
But it supposed to be?

Please provide me the actual results to identity the issue quicker.
Thanks

File: queries.sql [hunk 2]
```diff
@@ -105,3 +105,79 @@ WHERE o.full_name = 'Dean Winchester' AND (a.escape_attempts IS NULL OR a.escape
...
+-- How many visits were with a vet that did not specialize in that animal's species?
+SELECT count (*) FROM (
+  SELECT vets.name, species.name, vets.id
+  FROM vets LEFT JOIN specializations ON vets.id = specializations.vet_id 
+  LEFT JOIN species ON species.id = specializations.species_id
+  WHERE species.name IS NULL
+) AS no_specializations
+JOIN visits ON visits.vet_id = no_specializations.id;
```
Comment (rotimiazeez) [inline queries.sql]: Kindly update the screenshot result for this query as it should return 12

File: queries.sql [hunk 3]
```diff
@@ -105,3 +105,79 @@ WHERE o.full_name = 'Dean Winchester' AND (a.escape_attempts IS NULL OR a.escape
...
+INNER JOIN vets vt ON vt.id = v.vet_id
+WHERE vt.name = 'Stephanie Mendez' AND v.visit_date BETWEEN '2020-04-01' AND '2020-08-30';
+
+
+-- What animal has the most visits to vets?
+SELECT animals.name FROM (
+  SELECT count (*) as num_of_visit, animal_id
+  FROM visits GROUP BY animal_id ORDER BY num_of_visit DESC LIMIT 1
+  ) 
+top_visit JOIN animals ON top_visit.animal_id = animals.id;
```
Comment (rotimiazeez) [inline queries.sql:147]: Kindly update the screenshot result for this query as  the animal with the most visits to the vet should return `boarmon` which has 4 visit

**Review by rotimiazeez — COMMENTED** [review]
Hi @mahabubx7 😉, 

Great work on making the changes 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## Highlights
- Good communication with CR


## Suggested changes
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me(@rotimiazeez) in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid, unless it is requested otherwise._**

File: queries.sql (hunk 2 above)
Comment (rotimiazeez) [reply queries.sql]: Kindly update the screenshot result for the What animal has the most visits to vets? it should return `boarmon` with 4visits

File: queries.sql [hunk 4]
```diff
@@ -1,107 +1,287 @@
...
+SELECT count (*)
+FROM
+  (SELECT vets.name,
+          species.name,
+          vets.id
+   FROM vets
+   LEFT JOIN specializations ON vets.id = specializations.vet_id
+   LEFT JOIN species ON species.id = specializations.species_id
+   WHERE species.name IS NULL) AS no_specializations
+JOIN visits ON visits.vet_id = no_specializations.id;
```
Comment (t-yanick) [inline queries.sql]: - [ ] As indicated by the previous reviewer, the query result of this query should result in 12 and that should be put in the PR description. At the moment, what you have there is giving 9 as results. Kindly re-run this query and put a screenshot with the correct results (12)
image

**Review by t-yanick — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED

Hi @mahabubx7, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by amejid — APPROVED** [review]
Hi @mahabubx7, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Correct workflow ✔️ 
- SQL organized in multiple files ✔️ 
- Queries returning correct results ✔️ 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**